src/Module.ts

```diff
@@ -471,24 +470,33 @@ export default class Module {
 
 	private getOriginalLocation(
 		sourcemapChain: RawSourceMap[],
-		location: { line: number; column: number }
+		location: { line: number; column: number; source?: string; name?: string }
 	) {
-		const filteredSourcemapChain = sourcemapChain
-			.filter(sourcemap => sourcemap.mappings)
-			.map(sourcemap => {
-				const encodedSourcemap = sourcemap;
-				if (sourcemap.mappings) {
-					encodedSourcemap.mappings = encode(encodedSourcemap.mappings);
-				}
-				return encodedSourcemap;
-			});
+		const filteredSourcemapChain = sourcemapChain.filter(sourcemap => sourcemap.mappings);
+
 		while (filteredSourcemapChain.length > 0) {
 			const sourcemap = filteredSourcemapChain.pop();
-			const smc = new SourceMapConsumer(sourcemap);
-			location = smc.originalPositionFor({
-				line: location.line,
-				column: location.column
-			});
+			const line = sourcemap.mappings[location.line - 1];
+			let locationFound = false;
+
+			if (line !== undefined) {
+				for (const segment of line) {
```

Comment:
Could we make this `for (const segment of <[number, number, number, number][]>line)` to get the types to work out without the coersions then?

Reply:
I tried something like this but TypeScript errored with "can't coerce a string into an array of numbers".

Reply:
Then defining `line` above as `const line: any = ...` could work along with the for typing. Then we can further look at fixing this up in due course.